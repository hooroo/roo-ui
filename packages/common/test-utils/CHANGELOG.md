# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

<a name="0.4.0"></a>
# [0.4.0](https://github.com/hooroo/roo-ui/compare/v0.3.3...v0.4.0) (2018-04-26)


### Features

* add theming ([bee896a](https://github.com/hooroo/roo-ui/commit/bee896a))
