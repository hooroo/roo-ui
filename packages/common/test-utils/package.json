{
  "name": "@roo-ui/test-utils",
  "version": "0.0.0",
  "main": "dist/index.js",
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "build": "babel src -d dist --copy-files"
  }
}
