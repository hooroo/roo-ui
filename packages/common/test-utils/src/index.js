export { default as shallowWithTheme } from './shallowWithTheme'; // eslint-disable-line import/prefer-default-export
