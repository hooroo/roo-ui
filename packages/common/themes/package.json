{
  "name": "@roo-ui/themes",
  "version": "0.4.0",
  "main": "dist/index.js",
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "build": "babel src -d dist --copy-files"
  }
}
