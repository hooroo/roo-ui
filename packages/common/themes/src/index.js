import qantas from './qantas';

export { qantas }; // eslint-disable-line import/prefer-default-export
