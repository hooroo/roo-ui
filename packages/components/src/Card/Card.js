import { Box } from '..';

const Card = Box.extend``;

Card.defaultProps = {
  boxShadow: 'default',
  borderRadius: 'default',
  bg: 'white',
  p: 4,
};

Card.displayName = 'Card';

export default Card;
