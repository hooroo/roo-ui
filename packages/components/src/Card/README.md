# Card

<!-- STORY -->

## Installation

```shell
$ yarn add @roo-ui/components
```

## Example

```js
import { Card } from '@roo-ui/components';

export default (
  <Card>Hello world</Card>
);
```

## Customization

This component can be customized with [styled-system](https://github.com/jxnblk/styled-system) by passing props for [space](https://github.com/jxnblk/styled-system#space-responsive), [width](https://github.com/jxnblk/styled-system#width-responsive), [color](https://github.com/jxnblk/styled-system#color-responsive), [box shadow](https://github.com/jxnblk/styled-system#misc), or [border radius](https://github.com/jxnblk/styled-system#borders).
