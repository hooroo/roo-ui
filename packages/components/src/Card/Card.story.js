import React from 'react';
import { storiesOf } from '@storybook/react';
import { withDocs } from 'storybook-readme';

import { Box } from '..';
import Card from './Card';
import README from './README.md';

storiesOf('Components|Card', module)
  .addDecorator(withDocs(README))
  .add('default', () => (
    <Box p={5} bg="grey.3">
      <Card>
        Hello world
      </Card>
    </Box>
  ));
