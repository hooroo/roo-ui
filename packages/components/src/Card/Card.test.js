import React from 'react';
import { qantas as theme } from '@roo-ui/themes';
import { shallowWithTheme } from '@roo-ui/test-utils';
import { axe } from 'jest-axe';

import Card from '.';

describe('<Card />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowWithTheme(<Card>Hello world</Card>, theme);
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('has no accessibility errors', async () => {
    expect(await axe(wrapper.html())).toHaveNoViolations();
  });
});
