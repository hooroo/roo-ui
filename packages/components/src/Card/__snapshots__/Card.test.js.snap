// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> renders correctly 1`] = `
<StyledComponent
  bg="white"
  borderRadius="default"
  boxShadow="default"
  forwardedRef={null}
  p={4}
>
  Hello world
</StyledComponent>
`;
